core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/SigningStage.java

```diff
@@ -150,6 +153,20 @@ private void updateInterceptorContext(SdkHttpFullRequest request, ExecutionConte
         executionContext.interceptorContext(executionContext.interceptorContext().copy(b -> b.httpRequest(request)));
     }
 
+    /**
+     * We do not sign if the Auth SchemeId is smithy.api#noAuth.
+     *
+     * @return True if request should be signed, false if not.
+     */
+    private boolean shouldSign(SelectedAuthScheme<?> selectedAuthScheme) {
+        // TODO: Should this string be a constant somewhere. Similar logic is used in AuthSchemeInterceptors.
+        return "smithy.api#noAuth".equals(selectedAuthScheme.authSchemeOption().schemeId());
+
+        // Hmm, these would be equivalent too:
+        // return selectedAuthScheme.identityProvider() == null;
```

Comment:
I did not get the significance of this comment.

Reply:
Was meant to just show other ways of implementing this method if a reviewer preferred another version. Removed it in https://github.com/aws/aws-sdk-java-v2/pull/4144/commits/11386d18ab615d74a485b25f1107aa3fd7598054